clang/lib/Sema/SemaExprCXX.cpp

```diff
@@ -503,17 +503,16 @@ bool Sema::checkLiteralOperatorId(const CXXScopeSpec &SS,
     const IdentifierInfo *II = Name.Identifier;
     ReservedIdentifierStatus Status = II->isReserved(PP.getLangOpts());
     SourceLocation Loc = Name.getEndLoc();
-    if (!PP.getSourceManager().isInSystemHeader(Loc)) {
```

Comment:
It isn't clear to me that this check was necessary at all?  So I've removed it.  Both diags happen now without the check, and unconditionally on each-other.  They also share the `FixItHint`, which was done in a frightening-ish way.